src/wallet/wallet.cpp

```diff
@@ -4125,7 +4124,7 @@ std::shared_ptr<CWallet> CWallet::CreateWalletFromFile(interfaces::Chain& chain,
         }
     }
 
-    int prev_version = walletInstance->nWalletVersion;
+    int prev_version = walletInstance->GetVersion();
```

Comment:
Note: This GetVersion call looks like only change in this PR that actually affects locking behavior (GetVersion() call below just acquires cs_wallet) recursively.